src/main/java/com/bakdata/kafka/KafkaApplication.java

```diff
@@ -178,8 +178,8 @@ public ImprovedAdminClient createAdminClient() {
         final ImprovedAdminClient.ImprovedAdminClientBuilder adminClientBuilder = ImprovedAdminClient.builder()
             .properties(this.getKafkaProperties())
             .timeout(ADMIN_TIMEOUT);
-        if (nonNull(this.getSchemaRegistryUrl())) {
-            adminClientBuilder.schemaRegistryUrl(this.getSchemaRegistryUrl());
+        if (nonNull(this.schemaRegistryUrl)) {
```

Comment:
Why even have the if? you can also pass null to the builder method, so just always pass `this.schemaRegistryUrl`